lib/ansible/modules/files/unarchive.py

```diff
@@ -391,13 +395,24 @@ def is_unarchived(self):
                 ftype = 'f'
 
             # Some files may be storing FAT permissions, not Unix permissions
+            # For FAT permissions, we will use a base permissions set of 777 if the item is a directory or has the execute bit set.  Otherwise, 666.
+            #     This permission will then be modified by the system UMask.
+            # BSD always applies the Umask, even to Unix permissions.
+            # For Unix style permissions on Linux or Mac, we want to use them directly.
+            #     So we set the UMask for this file to zero.  That permission set will then be unchanged when calling _permstr_to_octal
+
             if len(permstr) == 6:
                 if path[-1] == '/':
                     permstr = 'rwxrwxrwx'
                 elif permstr == 'rwx---':
                     permstr = 'rwxrwxrwx'
                 else:
                     permstr = 'rw-rw-rw-'
+                file_umask=umask
+            elif 'bsd' in 'bsd' in systemtype.lower():
```

Comment:
This looks like a typo.  I think you just want `elif 'bsd' in systemtype.lower()`